pkg/infra/tower.go

```diff
@@ -97,14 +97,15 @@ func (t *Tower) Reshuffle() {
 }
 
 func (t *Tower) endOfDay() {
-	t.dayInfo.CurrDay += 1
 	for _, agent := range t.Agents {
 		agent := agent.BaseAgent()
 		agent.hpDecay(t.healthInfo)
 		agent.increaseAge()
 		agent.updateTreaties()
-		t.stateLog.LogUtility(t.dayInfo, agent.agentType, agent.utility(), agent.IsAlive())
+		t.stateLog.LogUtility(t.dayInfo, agent.agentType, agent.currentUtility, agent.IsAlive())
+		agent.currentUtility = 0
```

Comment:
```suggestion
		agent.utility = 0
```
Can have an `EndOfDay` function in agents and we move all of these state updates to it, although can be a different PR

Reply:
Would be nice with a separate PR if ok